text/0000-remove-ndebug.md

```diff
@@ -0,0 +1,45 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Remove official support for the `ndebug` config variable, replace the current usage of it with a
+more appropriate 'debug_assertions` compiler-provided config variable.
+
+# Motivation
+
+The usage of 'ndebug' to indicate a release build is a strange holdover from C/C++. It is not used
+much and is easy to forget about. Since it used like any other value passed to the `cfg` flag, it
+does not interact with other flags such as `-g` or `-O`.
+
+The only current users of `ndebug` are the implementations of the `debug_assert!` macro. At the
+time of this writing integer overflow checking is will also be controlled by this variable. Since
+the optimisation setting does not influence `ndebug`, this means that code that the user expects to
+be optimised will still contain the overflow checking logic. Similarly, `debug_assert!` invocations
+are not removed, contrary to what intuition should expect. Enabling optimisations should been seen
+as a request to make the user's code faster, removing `debug_assert!` and other checks seems like
+a natural consequence.
+
+# Detailed design
+
+The `debug_assertions` variable, the replacement for the `ndebug` variable, will be compiler
```

Comment:
Just clarifying, but by variable do you mean `cfg` directive? For example, I would be able to do:

```rust
if cfg!(debug_assertions) { /* super-expensive code */ }
```